pkg/source.go

```diff
@@ -135,3 +137,34 @@ func StandardizeManifest(manifest *model.Manifest) error {
 	}
 	return nil
 }
+
+// GenerateBuildInfoFile creates the BUILDINFO file, used by build scripts
+func GenerateBuildInfoFile(manifest *model.Manifest) error {
+	content := fmt.Sprintf(
+		`istio.io/pkg/version.buildVersion=%s
+istio.io/pkg/version.buildGitRevision=%s
+istio.io/pkg/version.buildStatus=Clean
+istio.io/pkg/version.buildTag=%s
+istio.io/pkg/version.buildHub=%s
+`,
+		manifest.Version,
+		manifest.Dependencies.Istio.Sha,
```

Comment:
This will give us the wrong version for dependencies from operator and CNI I think

Reply:
Well, currently only Istio build system makes use of BUILDINFO.

cni does not even have a version flag.

operator currently outputs something like `version.BuildInfo{Version:"unknown", GitRevision:"unknown", GolangVersion:"go1.13.4", BuildStatus:"unknown", GitTag:"unknown"}`

Reply:
> operator currently outputs something like version.BuildInfo{Version:"unknown", GitRevision:"unknown", GolangVersion:"go1.13.4", BuildStatus:"unknown", GitTag:"unknown"}

:slightly_frowning_face: they should be using it

Reply:
yeah. I can drop this and just remove the dirty/Modified checks in istio/istio.

I kind of like them though. I think they can be useful when people are building on their own (not useful to our official releases though).

Reply:
ok, we should make sure we update this to pass in a SHA if operator and/or CNI fix this. I will report issues for both

Reply:
CNI doesn't have a version. Operator recently fix this